CMakeLists.txt

```diff
@@ -1,10 +1,28 @@
 cmake_minimum_required (VERSION 3.5)
 
-project(msft_proxy)
+project(msft_proxy
+        VERSION 0.1.0 # local build version
+        LANGUAGES CXX)
 add_library(msft_proxy INTERFACE)
 target_compile_features(msft_proxy INTERFACE cxx_std_20)
-target_include_directories(msft_proxy INTERFACE .)
+target_include_directories(msft_proxy INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+                                                $<INSTALL_INTERFACE:include>)
 
+# install and export the project. project name - proxy
+include(GNUInstallDirs)

```

Comment:
Shall we do this on Windows as well?

Reply:
Yes, I'd suggest take the naming conventions from GNU even the repo is going to install on Windows.
We don't specify any absolute path in the install & export phase, so all the directories and file will get combined with `CMAKE_INSTALL_PREFIX` which is given by user during installation.


Details about GNUInstallDirs:
> Define GNU standard installation directories
> Provides install directory variables as defined by the [GNU Coding Standards](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html).